switchbot/devices/lock.py

```diff
@@ -165,6 +166,16 @@ async def lock(self) -> bool:
 
     async def unlock(self) -> bool:
         """Send unlock command."""
+        if (self.is_night_latch_enabled):
+            return await self._lock_unlock(
+                COMMAND_NOT_FULLY_UNLOCK, {LockStatus.UNLOCKED, LockStatus.UNLOCKING}
+            )
+        return await self._lock_unlock(
+            COMMAND_UNLOCK, {LockStatus.UNLOCKED, LockStatus.UNLOCKING, LockStatus.NOT_FULLY_LOCKED}
+        )
+
+    async def unlatch(self) -> bool:
+        """Send unlatch command. Unlatches door if feature is enabled elsewhere just unlock them."""
         return await self._lock_unlock(
             COMMAND_UNLOCK, {LockStatus.UNLOCKED, LockStatus.UNLOCKING}
         )
```

Comment:
Please make separate functions for each command.  We don't want conditionals in the functions.  The consumer of the api should call the correct one based on what they want, no what the lib thinks they want